static-site/components/list-resources/modal-contents-group-history.tsx

```diff
@@ -0,0 +1,179 @@
+  const filterWords = selectedFilterOptions
+    .map((opts) => opts.value)
+    .filter((value) => value!==group.groupName);
+  const filteredHistory = _filterHistory(history, filterWords);
+
+  const nDays = filteredHistory.length;
+  const allFiles = filteredHistory.flatMap((h) => Object.keys(h[1]));
+  const nFilesUnique = (new Set(allFiles)).size;
+
+  return (
+    <>
+      <div className={styles.title}>
+        {`All past versions of ${group.groupName} workflow files`}
+      </div>
+      {`${nFilesUnique} different filenames uploaded over ${nDays} separate days going back to ${filteredHistory.at(-1)?.[0]}.`}
+      {` In total there are ${allFiles.length} files listed here.`}
+      <div className={styles.newLine}>
+        {`Each link points to a versioned file associated with the specified day. The data shown here may be out of date by a day or two`}
+        {` and thus there could be more recent versions of a particular file uploaded since ${filteredHistory.at(0)?.[0]};`}
+        {` please use the link on the background page to guarantee you're getting the latest version of a particular file.`}
+      </div>
+      {filterWords.length!==0 && (
+        <div className={styles.newLine}>
+          {` Filtered on ${filterWords.length} keywords: ${filterWords.map((x) => `"${x}"`).join(', ')}.`}
+        </div>
+      )}
+
+      {filteredHistory.map(([date, info]) => {
+        return (
+          <>
+            <div className={styles.changelogSectionTitle}>
+              {date}
+            </div>
+            <ul>
+              {Object.entries(info).map(([filename, url]) => {
+                const displayFilename = filename.split('/').join(' / ');
+                return (
+                  <li className={styles.list} key={filename}>
+                    <a href={url}>{displayFilename}</a>
+                  </li>
+                )
+              })}
+            </ul>
+          </>
+        )
+      })}
+    </>
+  )
+}
+
+
+function _filterHistory(history: GroupHistory, filters: string[]): GroupHistory {
+  if (!filters.length) return history;
+  // Use a cache to speed things up as the same filenames are repeated many times
+  const cache: Map<string, boolean> = new Map(); // true: passes filter, false: exclude
+  return history.flatMap(([date, files]) => { // flatMap allows empty array returns to disappear
+    const filePairs = Object.entries(files)
+      .filter(([filename, ]) => {
+        if (!cache.has(filename)) {
+          const filewords = filename.split('/');
+          const passes = filters.map((w) => filewords.includes(w)).every((el) => el===true);
+          cache.set(filename, passes);
+        }
+        return cache.get(filename);
+      })
+    if (filePairs.length===0) return []; // no matches from this day
+    return [[date, Object.fromEntries(filePairs)]]
+  })
+}
+
+
+type GroupHistory = [
+  date: string,
+  {[nameInclFilename: string]: string}
+][]
+
```

Comment:
since there's a `./types` file for this component, probably better to put this there and import it.